pulp_deb/app/serializers/content_serializers.py

```diff
@@ -273,6 +277,38 @@ def from822(cls, data, **kwargs):
                 # also save the fields not in TRANSLATION_DICT
                 args["custom_fields"][k] = v
 
+        # Delete package fields with values of incorrect type
+        if "installed_size" in args and not args["installed_size"].strip().isdigit():
+            log.warn(
+                ("Removing 'Installed-Size' field from {} as it contains incorrect type.").format(
+                    args["package"]
+                )
+            )
+            del args["installed_size"]
+        if "essential" in args and not isinstance(args["essential"], bool):
```

Comment:
This currently does not work, since we need to do something similar as for the "multi_arch" field:
```suggestion
        if "essential" in args and args["essential"] not in [ value[1] for value in BOOL_CHOICES ]:
```
Note that `BOOL_CHOICES` will need to be imported since it is not part of the `BasePackage` class (like `MULTIARCH_CHOICES` is).

Reply:
Does the args["essential"] already contains the BOOL_CHOICES value or is it still having value from package? If it is from former then it already is boolean but if it from latter then we need to import BOOL_CHOICES.

Reply:
`args["essential"]` contains what was read from the `Packages` file using the `python-debian.deb822` library.
In my test it was `args["essential"] = 'yes'` which is a string and not a bool, so the field was dropped despite containing the correct value "yes".